src/pykeen/training/training_loop.py

```diff
@@ -393,6 +394,11 @@ def _train(  # noqa: C901
             raise ValueError('must set training instances before running _train()')
         if self.optimizer is None:
             raise ValueError('optimizer must be set before running _train()')
+        # When using early stopping models have to be saved separately at the best epoch, since the training loop will
+        # due to the patience continue to train after the best epoch and thus alter the model
+        if stopper is not None:
+            last_best_stopper_model_epoch = None
+            best_epoch_model_path = tempfile.NamedTemporaryFile()
```

Comment:
Is the temporary directory created during size probing? If yes, is it correctly cleared when an (out-of-memory) exception occurred in the body?

Reply:
It clears itself when it goes out of scope, but I think it's unnecessary to create it during size probing 👍